src/helpers/assets.ts

```diff
@@ -107,6 +103,15 @@ export const buildCoinsList = (
     emptyCollectibles
   );
 
+  if (!assets.length) {
+    return {
+      addedEth,
+      assets,
+      smallBalancesValue: 0,
+      totalBalancesValue: 0,
+    };
+  }
```

Comment:
@estrattonbailey continuing from my comment above...

i added this chunk of code because previously the value of `assets` returned by `addEthPlaceholder` would never be equal to `[]`, since the eth placeholder row got added in the case that there were no other tokens. since that logic is now changed, it's possible to have an empty asset array so I had to account for that here.